### PR TITLE
Add adc driver function for imxrt

### PR DESCRIPTION
**Pull request by HarrisonChoi750405**

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [3d63ce74ab48b1cf2f33e9a1a496154ff26584c7] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3d63ce74ab48b1cf2f33e9a1a496154ff26584c7] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [cc4fe7e5bdaa437354ce747cfef6d9cf48853395] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [cc4fe7e5bdaa437354ce747cfef6d9cf48853395] - Code Rule Check Result:
os/arch/arm/src/imxrt/imxrt_adc.c:733: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:734: ERROR: [IDT_M_TAB] please, no space before tabs

**Comment by seinfra** [comment]
Target : [c69fdb257bd4222c9a45fd46676af8764ce90bcf] - Code Rule Check Result:
os/arch/arm/src/imxrt/imxrt_adc.c:735: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:736: ERROR: [IDT_M_TAB] please, no space before tabs

**Comment by seinfra** [comment]
Target : [c69fdb257bd4222c9a45fd46676af8764ce90bcf] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d7d3dff9766642fd876850db050ab20f7206dcdc] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d7d3dff9766642fd876850db050ab20f7206dcdc] - Code Rule Check OK.

File: os/arch/arm/src/imxrt/imxrt_adc.c [hunk 1]
```diff
@@ -727,6 +730,13 @@ static int adc_ioctl(FAR struct adc_dev_s *dev, int cmd, unsigned long arg)
 		adc_startconv(priv, true);
 		break;
 
+	// TO-DO
```
Comment (Taejun-Kwon) [inline os/arch/arm/src/imxrt/imxrt_adc.c]: TizenRT doesn't support ANIOC_STOP, ANIOC_TRIGGER is there.
You should remove todo comment and update it with ANIOC_TRIGGER

File: os/arch/arm/src/imxrt/imxrt_adc.c [hunk 2]
```diff
@@ -62,6 +63,7 @@ struct imxrt_dev_s {
 	uint8_t current;			/* Current ADC channel being converted */
 
 	WDOG_ID work;				/* Supports the IRQ handling */
+	struct work_s work_int;	        /* Supports the IRQ handling */
```
Comment (drashti304) [inline os/arch/arm/src/imxrt/imxrt_adc.c]: Please correct indentation

File: os/arch/arm/src/imxrt/imxrt_adc.c [hunk 3]
```diff
@@ -754,6 +764,26 @@ static struct imxrt_dev_s g_adcpriv = {
 
 static struct adc_dev_s g_adcdev;
 
+static inline void imxrt_adc_pins_init(void)
```
Comment (drashti304) [inline os/arch/arm/src/imxrt/imxrt_adc.c]: I think we can remove inline from this function. It has 3 function call.
What is your opinion that?

**Pull request closed without merging**